File: zircon.jvm.examples/src/main/java/org/hexworks/zircon/examples/GraphicalTilesetExample.java
package org.hexworks.zircon.examples;

import org.hexworks.zircon.api.GraphicalTilesetResources;
import org.hexworks.zircon.api.SwingApplications;
import org.hexworks.zircon.api.application.AppConfig;
import org.hexworks.zircon.api.data.Position;
import org.hexworks.zircon.api.data.Size;
import org.hexworks.zircon.api.data.Tile;
import org.hexworks.zircon.api.grid.TileGrid;
import org.hexworks.zircon.api.resource.TilesetResource;

import java.util.Random;

public class GraphicalTilesetExample {

    private static final int TERMINAL_WIDTH = 50;
    private static final int TERMINAL_HEIGHT = 24;
    private static final Size SIZE = Size.create(TERMINAL_WIDTH, TERMINAL_HEIGHT);
    private static final TilesetResource TILESET = GraphicalTilesetResources.nethack16x16();
    private static final String[] NAMES = new String[]{
            "Giant ant",
            "Killer bee",
            "Fire ant",
            "Werewolf",
            "Dingo",
            "Hell hound pup",
            "Tiger",
            "Gremlin"
    };
    private static final Random RANDOM = new Random();

    public static void main(String[] args) {

        TileGrid tileGrid = SwingApplications.startTileGrid(
                AppConfig.newBuilder().withSize(SIZE).build());

        for (int row = 0; row < TERMINAL_HEIGHT; row++) {
            for (int col = 0; col < TERMINAL_WIDTH; col++) {
                final String name = NAMES[RANDOM.nextInt(NAMES.length)];
                tileGrid.draw(Tile.newBuilder()
                                .withName(name)
                                .withTileset(TILESET)
                                .buildGraphicalTile(),
                        Position.create(col, row));
            }
        }

    }

}

File: zircon.jvm.examples/src/main/kotlin/org/hexworks/zircon/examples/other/GraphicalTilesetExample.kt
package org.hexworks.zircon.examples.other

import org.hexworks.zircon.api.GraphicalTilesetResources.nethack16x16
import org.hexworks.zircon.api.SwingApplications.startTileGrid
import org.hexworks.zircon.api.application.AppConfig
import org.hexworks.zircon.api.data.Position
import org.hexworks.zircon.api.data.Size
import org.hexworks.zircon.api.data.Tile
import java.util.*

object GraphicalTilesetExample {
    private const val TERMINAL_WIDTH = 50
    private const val TERMINAL_HEIGHT = 24
    private val SIZE = Size.create(TERMINAL_WIDTH, TERMINAL_HEIGHT)
    private val TILESET = nethack16x16()
    private val NAMES = arrayOf(
        "Giant ant",
        "Killer bee",
        "Fire ant",
        "Werewolf",
        "Dingo",
        "Hell hound pup",
        "Tiger",
        "Gremlin"
    )
    private val RANDOM = Random()

    @JvmStatic
    fun main(args: Array<String>) {
        val tileGrid = startTileGrid(
            AppConfig.newBuilder().withSize(SIZE).build()
        )
        for (row in 0 until TERMINAL_HEIGHT) {
            for (col in 0 until TERMINAL_WIDTH) {
                val name = NAMES[RANDOM.nextInt(NAMES.size)]
                tileGrid.draw(
                    Tile.newBuilder()
                        .withName(name)
                        .withTileset(TILESET)
                        .buildGraphicalTile(),
                    Position.create(col, row)
                )
            }
        }
    }
}